analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -34,6 +34,27 @@ protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterNodeAction(CheckConstructorDeclaration, SyntaxKind.ConstructorDeclaration);
             context.RegisterNodeAction(CheckDestructorDeclaration, SyntaxKind.DestructorDeclaration);
+            context.RegisterNodeAction(CheckPrimaryConstructorRef, SyntaxKind.ClassDeclaration, SyntaxKindEx.RecordClassDeclaration);
+            context.RegisterNodeAction(CheckPrimaryConstructorVal, SyntaxKind.StructDeclaration, SyntaxKindEx.RecordStructDeclaration);
+        }
+
+        private static void CheckPrimaryConstructorRef(SonarSyntaxNodeReportingContext context) =>
+            CheckPrimaryConstructor(context, checkInitializedFieldOrProperty: false);
+
+        private static void CheckPrimaryConstructorVal(SonarSyntaxNodeReportingContext context) =>
+            CheckPrimaryConstructor(context, checkInitializedFieldOrProperty: true);
+
+        private static void CheckPrimaryConstructor(SonarSyntaxNodeReportingContext context, bool checkInitializedFieldOrProperty)
+        {
+            var typeDeclaration = (TypeDeclarationSyntax)context.Node;
+            if (TypeDeclarationSyntaxExtensions.ParameterList(typeDeclaration) is { Parameters.Count: 0 } parameterList)
```

Comment:
I see how the ambiguity between the implementation of `ParameterList` in the `TypeDeclarationSyntaxExtensions` and the one in the lightup led to using a de-sugared call.

I am a bit confused by the two methods being different, while targeting both `TypeDeclarationSyntax`:
- the extension method in `SonarAnalyzer.Extensions` deals with `class`, `struct`, `record class` and `record struct`, delegating to the right wrapper
- on the other hand, the extension method in `StyleCop.Analyzers.Lightup` calls the right compiled lambda

As far as I can tell, both seem to deal with all sub-types of `TypeDeclarationSyntax`, so it's quite confusing to have both of these options available. Can't we make them one?

Reply:
Resolved by reducing the visibility of the `TypeDeclarationExtensions` to internal. We should use `ParameterList` from `TypeDeclarationSyntaxExtensions`